internal/uploadmgr/gc.go

```diff
@@ -0,0 +1,96 @@
+package uploadmgr
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/util/sets"
+	toolscache "k8s.io/client-go/tools/cache"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+type bundleGC struct {
+	storageDir          string
+	storageSyncInterval time.Duration
+	cache               cache.Cache
+	log                 logr.Logger
+}
+
+func NewBundleGC(cache cache.Cache, storageDir string, storageSyncInterval time.Duration) manager.Runnable {
```

Comment:
nit: Would be great to get a comment here describing what this is for and how to use it.